pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -886,6 +886,158 @@ LANGUAGE SQL VOLATILE;
 COMMENT ON FUNCTION SCHEMA_PROM.reset_metric_retention_period(TEXT)
 IS 'resets the retention period for a specific metric to using the default';
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_metric_compression_setting(metric_name TEXT)
+RETURNS BOOLEAN
+AS $$
+DECLARE
+    can_compress boolean;
+    result boolean;
+    metric_table_name text;
+BEGIN
+    SELECT exists(select * from pg_proc where proname = 'compress_chunk')
+    INTO STRICT can_compress;
+
+    IF NOT can_compress THEN
+        RETURN FALSE;
+    END IF;
+
+    SELECT table_name 
+    INTO STRICT metric_table_name
+    FROM SCHEMA_CATALOG.get_or_create_metric_table_name(metric_name);
+
+    SELECT EXISTS ( 
+        SELECT FROM _timescaledb_config.bgw_policy_compress_chunks p
```

Comment:
This should be based on h.compressed_hypertable_id IS NOT NULL. Not on the bgw policy. This should now work given that we set the  timescaledb.compress = false in the off code.